ext/spl/tests/recursive_tree_iterator_007.phpt

```diff
@@ -22,12 +22,12 @@ try {
 	foreach(new RecursiveTreeIterator($it) as $k => $v) {
 		echo "[$k] => $v\n";
 	}
-} catch (UnexpectedValueException $e) {
-	echo "UnexpectedValueException thrown\n";
+} catch (Error $e) {
+	echo $e->getMessage(), "\n";
 }
 
 ?>
 ===DONE===
 --EXPECT--
-UnexpectedValueException thrown
```

Comment:
The test does not seem to cover anymore what it used to test. Also the comment in https://github.com/php/php-src/blob/f0aeadabf711650c6208180ec3826aca3414c186/ext/spl/spl_iterators.c#L1072-L1073 seems outdated now as it does not cause a catchable fatal error anymore.

Reply:
Thanks for pointing this out, I've dropped the now unnecessary code. I'm leaving the test though, as it will shows that that we're properly generating an exception here.